eng/DeterministicBuild.targets

```diff
@@ -0,0 +1,15 @@
+<!-- Allows Collection of Code Coverage for Deterministic Builds-->
```

Comment:
This seems more about enabling code coverage so perhaps we should name it that way and add in all the coverage stuff into this file and import it only if coverage is enabled.